### DISCUSSION TITLE
Is which fields string interpolation supported ?

### DISCUSSION THREAD

**Discussion by eshepelyuk** [Kuadrant/authorino#489] [Q&A] [answered]
Hello

For example, can one use _dynamic_ values in `jwt.issuerUrl`, e.g. retrieve it from HTTP request, like headers ?

**Comment by guicassolato** [comment, marked as answer]
@eshepelyuk

I'm afraid `spec.authentication.jwt.issuerUrl` does not support string interpolation. This is because Authorino will fetch the JSON Web Key Sets (JWKS) outside of an Auth Pipeline, where it does not count with the context required to resolve variables otherwise interpolated into the issuer's base URL.

Places in a AuthConfig where you can use string interpolation are:

`url` fields of:
- `spec.metadata..http`
- `spec.authorization..opa.externalPolicy`
- `spec.callback..http`

`selector` fields of:
- `spec.patterns`
- `spec.when` and `spec...when`
- `spec...cache.key`
- `spec.authentication..plain`
- `spec.authentication...defaults`
- `spec.authentication...overrides`
- `spec.metadata..http.body`
- `spec.metadata..http.bodyParameters`
- `spec.metadata..http.headers`
- `spec.authorization..kubernetesSubjectAccessReview.user`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.group`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.resource`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.subresource`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.name`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.namespace`
- `spec.authorization..kubernetesSubjectAccessReview.resourceAttributes.verb`
- `spec.authorization..opa.externalPolicy.body`
- `spec.authorization..opa.externalPolicy.bodyParameters`
- `spec.authorization..opa.externalPolicy.headers`
- `spec.authorization..patternMatching.patterns`
- `spec.authorization..spicedb.subject`
- `spec.authorization..spicedb.resource`
- `spec.authorization..spicedb.permission`
- `spec.response.success.headers..plain`
- `spec.response.success.headers..json.properties`
- `spec.response.success.headers..wristband.customClaims`
- `spec.response.success.dynamicMetadata..plain`
- `spec.response.success.dynamicMetadata..json.properties`
- `spec.response.success.dynamicMetadata..wristband.customClaims`
- `spec.response.unauthenticated.message`
- `spec.response.unauthenticated.body`
- `spec.response.unauthorized.message`
- `spec.response.unauthorized.body`
- `spec.callback..http.body`
- `spec.callback..http.bodyParameters`
- `spec.callback..http.headers`

Apologies in advance if I missed some.

**Comment by eshepelyuk** [comment]
Thank you
